src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -47,7 +50,57 @@ public static LoginData performUserLogin() {
         Popups.showMessageSync("Login", pane, panel::setDialog);
 
         LoginData loginData = panel.getResult();
-        if (loginData.getPreloaderUrl() == null || loginData.getParams() == null) System.exit(0);
+        if (loginData.getPreloaderUrl() == null || loginData.getParams() == null) {
+            System.err.println("Could not find preloader url or parameters");
+            System.exit(-1);
```

Comment:
I'm pretty sure in this case, it means the user closed the panel to login without selecting a user or logging in in any other way, it's not an error to shutdown with -1, rather, 0, and make the it a System.out instead of err, but mention how the user exited without logging in, not that preloader is missing